src/core/agent/services/connectionService.ts

```diff
@@ -65,7 +66,23 @@ class ConnectionService extends AgentService {
   onConnectionStateChanged(
     callback: (event: ConnectionStateChangedEvent) => void
   ) {
-    this.props.eventEmitter.on(EventTypes.ConnectionStateChanged, callback);
+    this.props.eventEmitter.on(
+      EventTypes.ConnectionStateChanged,
+      (event: ConnectionStateChangedEvent) => {
+        if (event.payload.url) {
+          this.resolveOobi(event.payload.url);
+        }
```

Comment:
I would leave this handler simple. The UI calls `onConnectionStateChanged` with their callback to update Redux.

But we can register a second handler in the code with our own handler, that calls `resolveOobi`

Reply:
This separation is better and clearer